### DISCUSSION TITLE
How to restrict the client's access to the modbus TCP server

### DISCUSSION THREAD

**Discussion by Jan5566** [pymodbus-dev/pymodbus#2067] [Help] [answered]
Is that possible to get client IP when use modbus TCP server? The purpose is to limit which can access registers by IP. Restrict which IP can read/write, and which only can read

**Comment by janiversen** [comment]
There are no API for that since the server do not have an API apart from startup and shutdown. the transport layer have that knowledge since it's a socket.

**Comment by janiversen** [comment]
The restriction itself is already in place in the datastore but needs to be amended.

**Reply by Jan5566** [reply]
Thanks for the reply, could you provide some examples?

**Reply by janiversen** [reply, marked as answer]
No I have never done that kind of limitation, but the best examples are to look at the source.
